### PR TITLE
ci: add nightly e2e test workflow

### PR DESCRIPTION
**Pull request by cmwaters**
Adds a workflow to run all e2e test once a day

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: stale
```
Comment (rootulp) [inline .github/workflows/nightly.yml:1]:
```suggestion
name: nightly
```

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  schedule:
+    # runs every three days at 6am UTC
+    - cron: "0 6 */3 * *"
```
Comment (rootulp) [inline .github/workflows/nightly.yml:5]: [question] do the e2e tests take less than 24 hours to run? If yes, is it possible to run these every 24 hours?

**Review by rootulp — COMMENTED** [review]
🚀

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (cmwaters) [reply .github/workflows/nightly.yml:1]: lol good old copy paste

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (cmwaters) [reply .github/workflows/nightly.yml:5]: Yeah let's do it daily

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+      - name: Setup kubeconfig
+        env:
+          KUBECONFIG_FILE: ${{ secrets.KNUU_KUBECONFIG_FILE }}
+        run: |
+          mkdir -p $HOME/.kube
+          echo "${KUBECONFIG_FILE}" > $HOME/.kube/config
+
+      - name: Run e2e tests
+        run: make test-e2e
```
Comment (evan-forbes) [inline .github/workflows/nightly.yml:28]: per linter
```suggestion
        run: make test-e2e

```

**Review by evan-forbes — APPROVED** [review]
are the results of this nightly test just in the CI actions tab on github?

**Review by evan-forbes — APPROVED** [review]

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -1,8 +1,8 @@
-name: stale
+name: nightly
 on:
   schedule:
     # runs every three days at 6am UTC
-    - cron: "0 6 */3 * *"
+    - cron: "0 6 * * *"
```
Comment (rootulp) [inline .github/workflows/nightly.yml:5]: 👍  https://crontab.guru/#0_6_*_*_*

File: .github/workflows/nightly.yml [hunk 5]
```diff
@@ -1,8 +1,8 @@
-name: stale
+name: nightly
 on:
   schedule:
     # runs every three days at 6am UTC
```
Comment (rootulp) [inline .github/workflows/nightly.yml:4]:
```suggestion
    # runs every day at 6am UTC
```

**Review by rootulp — CHANGES_REQUESTED** [review]

**Review by rootulp — APPROVED** [review]

File: Makefile [hunk 6]
```diff
@@ -149,8 +149,9 @@ test-short:
 
 ## test-e2e: Run end to end tests via knuu.
 test-e2e:
-	@echo "--> Running e2e tests"
-	@KNUU_NAMESPACE=celestia-app E2E=true go test ./test/e2e/... -timeout 30m
+	@version=$(git rev-parse --short HEAD)
+	@echo "--> Running e2e tests on version: $version"
+	@KNUU_NAMESPACE=test E2E_VERSION=$version E2E=true go test ./test/e2e/... -timeout 30m
```
Comment (smuu) [inline Makefile:154]: You can use your namespace instead to be able to figure out your logs and metrics more easily.

```suggestion
	@KNUU_NAMESPACE=knuu-app E2E_VERSION=$version E2E=true go test ./test/e2e/... -timeout 30m
```

**Review by mojtaba-esk — APPROVED** [review]

**Comment by cmwaters** [comment]
> are the results of this nightly test just in the CI actions tab on github?

I think it would be nice to add a hook to slack on failure, but I thought of doing this in a different PR and once we know it's working as expected (so I can avoid spamming people for a few days)

**Comment by cmwaters** [comment]
> Another idea would be to run this workflow when you create a new tag so it runs when you create a new rc.

Yeah that makes sense. Maybe we just make it a manual workflow that we can trigger before cutting the release

**Pull request merged**